### PR TITLE
Agent documentation.

### LINKED ISSUE #1389: Document agent setup
Currently, in the documentation under installation there is a chapter on the Inmanta server.
A second chapter on different ways to set up agents would be a natural second chapter to go with it.

### PR DESCRIPTION
**Pull request by arnaudsjs**
Fixes #1389

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+-----------------------------------
 
 Logging can be configured by following the instructions in :ref:`administrators_doc_logging`.
+
+
+Configure agents
+################
+
+Inmanta agents can be started automatically (auto-started agents) or manually (manually-started agents). This section
+describes how both types of agents can be set up and configured.
```
Comment (kazaamjt) [inline docs/install.rst]: Mention agents are Linux only.

File: docs/install.rst [hunk 2]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+* autostart_agent_map
+* autostart_agent_deploy_interval
+* autostart_agent_deploy_splay_time
+* autostart_agent_repair_interval
+* autostart_agent_repair_splay_time
+* autostart_on_start
+
+The ``autostarted_agent_map`` requires an entry for each agent that should be autostarted. The key is the name of the agent and
+the value is either ``local:`` if the handlers should be executed on the Inmanta server or an SSH connection string when the
+handlers should be executed on a remote machine.The SSH connection string requires the following format:
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
handlers should be executed on a remote machine. The SSH connection string requires the following format:
```
add a space

File: docs/install.rst [hunk 3]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+``ssh://<user>@<host>:<port>?<options>``. Options is a ampersand-separated list of ``key=value`` pairs. The following options
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
``ssh://<user>@<host>:<port>?<options>``. Options is an ampersand-separated list of ``key=value`` pairs. The following options
```

File: docs/install.rst [hunk 4]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+^^^^^^^^^^^^
+
+If the handler code should be executed on another machine than the Inmanta server, the following requirements should be met:
+
+* The Inmanta server should have passphraseless SSH access on the remote machine. More information on how to set up SSH
+  connectivity can be found at :ref:`configure_server_step_6`
+* The remote machine should have a Python interpreter installed.
+
+
+Configure auto-started agents via environment settings
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
Configuring auto-started agents via environment settings
```

File: docs/install.rst [hunk 5]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+retry_wait   30             The amount of second between two attempts to establish the SSH connection.
+python       python         The Python interpreter available on the remote side.
+===========  =============  ==============================================================================================================
+
+
+Auto-started agents start when they are required by a specific deployment or when the Inmanta server starts if the
+``autostart_on_start`` setting is set to true.
+
+
+Configure the autostart_agent_map via the std::AgentConfig entity
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
Configuring the autostart_agent_map via the std::AgentConfig entity
```

File: docs/install.rst [hunk 6]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+If the handler code should be executed another machine than where the agent is running, the following requirements should be
+met:
+
+* The Inmanta agent should have passphraseless SSH access on the remote machine. More information on how to set up SSH
+  connectivity can be found at :ref:`configure_server_step_6`
+* The remote machine should have a Python interpreter installed.
+
+
+
+Step 1: Install the required Inmanta packages
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
Step 1: Installing the required Inmanta packages
```

File: docs/install.rst [hunk 7]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+    gpgkey=https://pkg.inmanta.com/inmanta-oss-stable/inmanta-oss-stable-public-key
+    repo_gpgcheck=1
+    enabled=1
+    enabled_metadata=1
+    EOF
+
+    sudo yum install -y python3-inmanta python3-inmanta-agent
+
+
+Step 2: Configure the manually-started agent
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
Step 2: Configuring the manually-started agent
```

File: docs/install.rst [hunk 8]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+handlers should be executed on a remote machine.The SSH connection string requires the following format:
+``ssh://<user>@<host>:<port>?<options>``. Options is a ampersand-separated list of ``key=value`` pairs. The following options
+can be provided:
+
+===========  =============  ==============================================================================================================
+Option name  Default value  Description
+===========  =============  ==============================================================================================================
+retries      10             The amount of times the orchestrator will try to establish the SSH connection when the initial attempt failed.
+retry_wait   30             The amount of second between two attempts to establish the SSH connection.
+python       python         The Python interpreter available on the remote side.
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
python       python         Path to the Python interpreter available on the remote side.
```

File: docs/install.rst [hunk 9]
```diff
@@ -250,13 +252,157 @@ Step 8: Connect to the dashboard
...
+* :inmanta.config:option:`agent_rest_transport.token`
+* :inmanta.config:option:`agent_rest_transport.ssl`
+* :inmanta.config:option:`agent_rest_transport.ssl-ca-cert-file`
+
+
+The :inmanta.config:option:`config.agent-map` option can be configured in the same way as the ``autostart_agent_map`` for
+auto-started agents.
+
+
+Step 3: Start the manually-started agent
```
Comment (kazaamjt) [inline docs/install.rst]:
```suggestion
Step 3: Starting the manually-started agent
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -8,6 +8,7 @@ v 2019.4 (?) Changes in this release:
 - Split out the main service into many smaller services (#1388)
 - Use python3 from the core OS in Dockerfile
 - Introduce v2 protocol and implement project and environment api in v2 (#1412)
+- Add agent documentation (#1389)
```
Comment (bartv) [inline CHANGELOG.md]:
```suggestion
- Improve agent documentation (#1389)
```

File: docs/install/2-configure-agents.rst [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+
+Requirements
+============
+
+The following requirements should be met for agents that don't map to the Inmanta server (i.e. The managed device is remote
+with respect to the Inmanta server and the agent has to execute I/O operations on the remote machine using ``self._io``):
+
+* The Inmanta server should have passphraseless SSH access on the machine it maps to. More information on how to set up SSH
+  connectivity can be found at :ref:`configure_server_step_6`
+* The remote machine should have a Python2 interpreter installed.
```
Comment (bartv) [inline docs/install/2-configure-agents.rst]:
```suggestion
* The remote machine should have a Python 2 or 3 interpreter installed. The default binary is executed is python.
```

File: docs/install/2-configure-agents.rst [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+Requirements
+============
+
+The following requirements should be met for agents that don't map to the host running the agent process (i.e. The managed
+device is remote with respect to the Inmanta agent and the agent has to execute I/O operations on the remote machine using
+``self._io``):
+
+* The Inmanta agent should have passphraseless SSH access on the machine it maps to. More information on how to set up SSH
+  connectivity can be found at :ref:`configure_server_step_6`
+* The remote machine should have a Python2 interpreter installed.
```
Comment (bartv) [inline docs/install/2-configure-agents.rst]:
```suggestion
* The remote machine should have a Python 2 or 3 interpreter installed. The default binary is executed is python.```

**Pull request merged**